lib/java/com/google/android/material/textfield/TextInputLayout.java

```diff
@@ -313,8 +313,13 @@ public TextInputLayout(Context context, AttributeSet attrs, int defStyleAttr) {
       defaultHintTextColor =
           focusedTextColor = a.getColorStateList(R.styleable.TextInputLayout_android_textColorHint);
     }
-    defaultStrokeColor =
-        ContextCompat.getColor(context, R.color.mtrl_textinput_default_box_stroke_color);
+    if (a.hasValue(R.styleable.TextInputLayout_defaultBoxStrokeColor)) {
+      defaultStrokeColor =
+          a.getColor(R.styleable.TextInputLayout_defaultBoxStrokeColor, Color.TRANSPARENT);
+    } else {
+      defaultStrokeColor =
+          ContextCompat.getColor(context, R.color.mtrl_textinput_default_box_stroke_color);
+    }
```

Comment:
Any reason you do the hasValue() check instead of using the default value parameter?

    defaultStrokeColor =
        a.getColor(
            R.styleable.TextInputLayout_defaultBoxStrokeColor,
            ContextCompat.getColor(context, R.color.mtrl_textinput_default_box_stroke_color));

Reply:
This approach avoids the computation of the getColor, but I can remove it since your approach is easier to maintain.